components/timezone/src/ids.rs

```diff
@@ -97,14 +100,21 @@ impl TimeZoneIdMapper {
 
     /// Returns a borrowed version of the mapper that can be queried.
     ///
-    /// This avoids a small potential cost of reading the data pointer.
+    /// This avoids a small potential indirection cost when querying the mapper.
     pub fn as_borrowed(&self) -> TimeZoneIdMapperBorrowed {
         TimeZoneIdMapperBorrowed {
             data: self.data.get(),
         }
     }
 }
 
+impl AsRef<TimeZoneIdMapper> for TimeZoneIdMapper {
```

Comment:
whatever you need this for, can you use `Borrow` which has a blanket impl?

Reply:
I need it for the

```rust
impl<I> TimeZoneIdMapperWithFastCanonicalization<I>
where
    I: AsRef<TimeZoneIdMapper>,
```

`AsRef` is more general than `Borrow`.

I think we discussed this before, in the context of PluralRulesWithRanges, but I don't recall us reaching a clear conclusion.

Reply:
Yeah it was discussed here, with `AsRef`: https://github.com/unicode-org/icu4x/issues/4140